src/app-logic/web-channel.js

```diff
@@ -163,6 +259,25 @@ function _sendMessageWithResponse<Returns: MessageFromBrowser>(
 
     window.addEventListener('WebChannelMessageToContent', listener, true);
 
+    // Add the requestId to the message. The (...: any) is due to
+    // https://github.com/facebook/flow/issues/1488 which is probably fixed
+    // in flow v0.111.
+    const messageToBrowser: MessageToBrowser<B> = ({
+      requestId,
+      ...messageToBrowserBody,
+    }: any);
+
     _sendMessage(messageToBrowser);
   });
 }
+
+function _fixupOldResponseMessageIfNeeded(message: MixedObject) {
```

Comment:
nit: please add a comment explaining again for which versions of firefox this is needed, as well as the possible inputs